client/src/main/java/io/pravega/client/stream/impl/EventStreamWriterImpl.java

```diff
@@ -332,24 +332,26 @@ private void throwIfClosed() throws TxnFailedException {
     public void flush() {
         Preconditions.checkState(!closed.get());
         synchronized (writeFlushLock) {
-            flushInternal();
+            boolean success = false;
+            while (!success) {
+                success = flushInternal();
+            }
         }
     }
 
-    private void flushInternal() {
-        boolean success = false;
-        while (!success) {
-            success = true;
-            for (SegmentOutputStream writer : selector.getWriters()) {
-                try {
-                    writer.flush();
-                } catch (SegmentSealedException e) {
-                    // Segment sealed exception observed during a flush. Re-run flush on all the available writers.
-                    success = false;
-                    log.warn("Flush failed due to {}, it will be retried.", e.getMessage());
-                }
+    private boolean flushInternal() {
```

Comment:
`Observation`: In the older code state.waitForInflight implied inflight events are empty. Now it is not the case. 

Details:
```
T-nx --> Netty thread.
T-cfx --> Connection factory thread.
T-rx --> RetransmitPool thread.
Q --> Query
```
If we assume a there are two segments `S0` and `S1` and S1 sealed callback is received.
```
`T-n1` ---> processes SegmentSealedCallback --> S1.state.sealEncountered= true -->`T-cf1` --> 
invokes handleLogSealed() --> add to segmentSealedQueue --> `T-r1` --> 
resend(inflight events of S1) --> invoke flush internal --> invoke S0.flush() --> this has inflight 
--> waits until inflight empty.
```
Now, if S0 also encounters segment sealed.
`T-n2` --> SegmentSealedCallback --> S0.state.sealEncountered=true --> S0.state.waitingInflight.released()->
 (This would cause the S0.flush() (`T-r1`) to complete without actually re transmitting the inflight events (it throws SegSealedException))
--> `T-cf2` -- invoke handleLogSealed() 
(`Q`: This thread T-cf2 could be starved and cause the resend of inflight to be delayed, right? )

(is the below scenario even valid?)
Also, if a reconnect() invoked during this interval it would cause the `state.isAlreadySealed()` to return true causing future writes to receive IllegalStateException right?